modules/phase_field/src/postprocessors/FeatureFloodCount.C

```diff
@@ -330,32 +331,138 @@ FeatureFloodCount::communicateAndMerge()
    * However, We do know the number of incoming buffers (num processors) so we'll
    * go ahead and use a vector.
    */
-  std::vector<std::string> recv_buffers;
-  if (_is_master)
-    recv_buffers.reserve(_app.n_processors());
-
-  serialize(send_buffers[0]);
-
-  // Free up as much memory as possible here before we do global communication
-  clearDataStructures();
+  std::vector<std::string> recv_buffers, deserialize_buffers;
 
   /**
-   * Send the data from all processors to the root to create a complete
-   * global feature map.
+   * When we distribute merge work, we are reducing computational work by adding more communication.
+   * Each of the first _n_vars processors will receive one variable worth of information to merge.
+   * After each of those processors has merged that information, it'll be sent to the master
+   * processor where final consolidation will occur.
    */
-  _communicator.gather_packed_range(0,
-                                    (void *)(nullptr),
-                                    send_buffers.begin(),
-                                    send_buffers.end(),
-                                    std::back_inserter(recv_buffers));
+  if (_distribute_merge_work)
+  {
+    auto rank = processor_id();
+    bool is_merging_processor = rank < _n_vars;
 
-  if (_is_master)
+    if (is_merging_processor)
+      recv_buffers.reserve(_app.n_processors());
+
+    for (auto i = decltype(_n_vars)(0); i < _n_vars; ++i)
+    {
+      serialize(send_buffers[0], i);
+
+      /**
+       * Send the data from all processors to the first _n_vars processors to create a complete
+       * global feature maps for each variable.
+       */
+      _communicator.gather_packed_range(i,
+                                        (void *)(nullptr),
+                                        send_buffers.begin(),
+                                        send_buffers.end(),
+                                        std::back_inserter(recv_buffers));
+
+      /**
+       * A call to gather_packed_range seems to populate the receiving buffer on all processors, not
```

Comment:
@roystgnr - There's a bug here in gather_packed_range somewhere. I get garbage in `recv_buffers` on processors that "don't" receive so I have to clean it up if I plan to reuse those buffers. I just noticed that my comment is a little misleading. I'm definitely not getting data on all processors (e.g. allgather), but the buffer isn't empty either. I'll see if I can work up a test case or something.